services/QuillLMS/spec/models/classroom_unit_spec.rb

```diff
@@ -142,6 +142,40 @@
           expect(other_classroom_unit.reload.assign_on_join).not_to be(true)
         end
       end
+
+      context 'when assign_on_join is explicitly set to false' do
```

Comment:
I think you pull out:
```
let(:assigned_student_ids) { [student.id, student2.id] }
let(:assign_on_join) { false }
```
which will make clear what's really varying test to test.

Reply:
This is actually probably a good call, but would involve rewriting most of the tests in the containing context block to use `expect.to change` patterns instead of the current `expect(pre_condition), update, expect(post_condition)`.

This may be worth doing, but I wanted to avoid touching working specs here.  Thoughts on whether it's worth it to overhaul the section to be more in line with our more state-of-the-art spec approach?

Reply:
You're right.  My mistake.